examples/password-manager/src/3_vault.tsx

```diff
@@ -17,20 +21,19 @@ const VaultPage: React.FC = () => {
   const { me, logOut } = useAccount();
   const sharedFolderId = useParams<{ sharedFolderId: ID<Folder> }>()
     .sharedFolderId;
-  const sharedFolder = useCoState(Folder, sharedFolderId);
+
   const navigate = useNavigate();
 
   useEffect(() => {
-    if (!sharedFolderId || !sharedFolder || !me.root?.folders) return;
-    const existsIndex = me.root?.folders.findIndex(
-      (f) => f?.id === sharedFolder.id
-    );
-    if (existsIndex > -1) {
-      me.root?.folders?.splice(existsIndex, 1);
-    }
-    me.root?.folders?.push(sharedFolder);
-    navigate("/vault");
-  }, [sharedFolder, me.root?.folders, sharedFolderId, navigate]);
```

Comment:
The `me.root.folders` dependency here now creates an infinite loop because the folders reference changes when the list changes.

Refactored to trigger the effect only on mount and moved the subscription logic into the action.